integrations/mcp/src/haystack_integrations/tools/mcp/mcp_tool.py

```diff
@@ -485,6 +498,13 @@ def __init__(
         self.token: str | None = (
             server_info.token.resolve_value() if isinstance(server_info.token, Secret) else server_info.token
         )
+        # Resolve Secret values in headers dictionary
+        self.headers: dict[str, str] | None = None
+        if server_info.headers:
+            self.headers = {
+                key: (value.resolve_value() if isinstance(value, Secret) else value) or ""
+                for key, value in server_info.headers.items()
+            }
```

Comment:
This logic seems to be the same in `SSEClient` and `StreamableHttpClient` - WDYT about extracting a common util method?